### PR TITLE
fix sampling, add error messages, fix random seeds

### PR DESCRIPTION
**Pull request by antschum**
Fix subsampling. 
Add error messages. 
Fix random seed.
Update tests accordingly.

### CODE REVIEW

File: augurpy/evaluate.py [hunk 1]
```diff
@@ -88,28 +88,51 @@ def draw_subsample(
...
-        stratified: if `True` subsamples are stratified according to condition
+        categorical_data: if `True` subsamples are
         random_state: set numpy random seed and sampling seed
 
     Returns:
         Subsample of anndata object of size subsample_size
     """
     if augur_mode == "permut":
         # shuffle labels
+        adata = adata.copy()
```
Comment (Zethson) [inline augurpy/evaluate.py:99]: Always an expensive operation and something that we try to avoid, but guess it's fine here.

File: augurpy/evaluate.py [hunk 2]
```diff
@@ -138,6 +161,24 @@ def ccc_score(y_true, y_pred) -> float:
     return ccc
 
 
+def tp_score(y_true, y_pred) -> float:
```
Comment (Zethson) [inline augurpy/evaluate.py]: Can't we use the sklearn implementation?

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html

File: augurpy/evaluate.py [hunk 3]
```diff
@@ -138,6 +161,24 @@ def ccc_score(y_true, y_pred) -> float:
...
+    """Implementation of true positive score."""
+    tp = 0
+    for i in range(len(y_true)):
+        if y_true[i] == y_pred[i] == 1:
+            tp += 1
+    return tp
+
+
+def fp_score(y_true, y_pred) -> float:
```
Comment (Zethson) [inline augurpy/evaluate.py]: Does sklearn have an implementation of this? Didn't see one on a quick glance, but would be surprised if it weren't there somewhere :)

**Review by Zethson — APPROVED** [review]
Very nice. LGTM!

File: augurpy/evaluate.py (hunk 1 above)
Comment (antschum) [reply augurpy/evaluate.py:99]: The reason I have to copy it is because Im changing the obs labels one line later, and it was changing them for the whole dataframe. I thought about only shuffling the labels of the already merged subsample adata object, but then the sampling would have been done based on the original labels ?

Comment (Zethson) [reply augurpy/evaluate.py:99]: No worries. It should be fine.

File: augurpy/evaluate.py (hunk 2 above)
Comment (antschum) [reply augurpy/evaluate.py]: Ahh I forgot that I had added these. I would actually vote to remove them again - this was when I was trying to figure out why precision was throwing an error. 

As to sklearn: I wanted the absolute counts which is not implemented in skelarn. The precision score itself is also added.

Comment (Zethson) [reply augurpy/evaluate.py]: All right, then let's remove them :)
